linera-sdk-derive/src/lib.rs

```diff
@@ -69,23 +75,47 @@ fn generate_mutation_root_code(input: ItemEnum) -> TokenStream2 {
         };
     }
 
-    quote! {
-        pub struct #mutation_root_name;
+    if in_crate {
```

Comment:
Could we create a `let path_root = if in_crate { quote! { crate } } else { linera_sdk };` instead?

I actually think it would be more idiomatic to have an attribute that overrides this, so that one could do:

```rust
#[derive(GraphQLMutationRoot)]
#[linera(crate_root = "crate")]
```

Reply:
Nice! Let me try

Reply:
I did it with the argument, but happy to try with the attribute also, with some guidance though 😅